### PR TITLE
add space between ingredient amount and unit

### PR DESCRIPTION
**Pull request by antomax33**

### CODE REVIEW

File: app/src/main/java/com/github/se/polyfit/ui/components/ingredients/IngredientList.kt [hunk 1]
```diff
@@ -143,8 +144,12 @@ private fun IngredientButton(ingredient: Ingredient, mealViewModel: MealViewMode
         onIngredientRemove = { mealViewModel.removeIngredient(ingredient) },
         onCollapseIngredient = { isExpanded = false })
   } else {
+
+    var text = "${ingredient.name} ${ingredient.amount}"
+    if (ingredient.unit != MeasurementUnit.OTHER) text += ingredient.unit
```
[the remark this answers is not in the document]
Comment (antomax33) [reply app/src/main/java/com/github/se/polyfit/ui/components/ingredients/IngredientList.kt]: :+1:

File: app/src/androidTest/java/com/github/se/polyfit/ui/screen/IngredientTest.kt [hunk 2]
```diff
@@ -86,6 +86,7 @@ class IngredientTest : TestCase() {
           Ingredient("White Asparagus", 10, 10.0, MeasurementUnit.G),
           Ingredient("Corn", 4, 4.0, MeasurementUnit.G),
           Ingredient("Foie Gras", 100, 100.0, MeasurementUnit.G),
+          Ingredient("Asperge", 27, 1.0, MeasurementUnit.OTHER),
```
Comment (Neb1010) [inline app/src/androidTest/java/com/github/se/polyfit/ui/screen/IngredientTest.kt:89]:
```suggestion
          Ingredient("Aspargus", 27, 1.0, MeasurementUnit.OTHER),
```

why for White Aspargus this is `MeasurementUnit.G` but for Aspargus it's `MeasurementUnit.Other` ?

Comment (antomax33) [reply app/src/androidTest/java/com/github/se/polyfit/ui/screen/IngredientTest.kt:89]: I wanted to test the Measurement Other. It doesn't display the same way.

File: app/src/main/java/com/github/se/polyfit/model/ingredient/Ingredient.kt [hunk 3]
```diff
@@ -28,8 +28,17 @@ data class Ingredient(
     return Ingredient(name, id, amount, unit, nutritionalInformation.deepCopy())
   }
 
+  private fun isInteger(d: Double): Boolean {
+    return (d == d.toInt().toDouble())
+  }
+
   fun amountFormatted(): String {
-    return amount.toString() + " " + unit.toString().lowercase(Locale.ROOT)
+    Log.d("Ingredient", "displayInformation: $name, amount: $amount, unit: $unit")
```
Comment (RandomUsername1315) [inline app/src/main/java/com/github/se/polyfit/model/ingredient/Ingredient.kt:36]:
```suggestion
```

**Review by RandomUsername1315 — APPROVED** [review]
Looks good !

**Pull request merged**